File: frontend/src/app/model/vendedor/vendedor.ts
import { Cliente } from "../cliente/cliente";

export class Vendedor {
    id: number;
    nome: String;
    email: String;
    senha: String;
    cliente?: Cliente;

    constructor(id: number, nome: String, email: String, senha: String, cliente?: Cliente) {
        this.id = id;
        this.nome = nome;
        this.email = email;
        this.senha = senha;
        this.cliente = cliente;
    }
}

File: backend/src/main/java/com/luis/authlogin/domain/vendedor/Vendedor.java
package com.luis.authlogin.domain.vendedor;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.luis.authlogin.domain.cliente.Cliente;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Table(name = "vendedores")
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Vendedor {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String nome;
    private String email;
    private String senha;

    @JsonManagedReference
    @OneToMany(mappedBy = "vendedor")
    private List<Cliente> clientes;

    public void addCliente(Cliente cliente) {
        this.clientes.add(cliente);
    }


}
